### ISSUE TITLE
[Bug]: No tests detected on Windows

### ISSUE THREAD

**Issue by ahmedneilhussain** [aspect-build/rules_jest#270]
### What happened?

I've been trying to get some simple unit tests working with `rules_jest`: it's been fine on Linux and OSX, but the rules fail on Windows: I just get

```
No tests found, exiting with code 1
Run with `--passWithNoTests` to exit with code 0
No files found in C:\Users\vagrant\_bazel_vagrant\x57pwzg3\execroot\_main\bazel-out\x64_windows-fastbuild\bin\src\src_test.bat.runfiles\_main\src
```

and variations on that theme. After struggling with my own examples, I tried to just run the example at https://github.com/aspect-build/bazel-examples/tree/main/jest

This showed the same problem.

### Version

Development (host) and target OS/architectures:
Windows 10 Pro

Output of `bazel --version`:

Version of the Aspect rules, or other relevant rules from your
`WORKSPACE` or `MODULE.bazel` file:

Language(s) and/or frameworks involved:

### How to reproduce

```shell
Just run `bazel run //src:src_test` on windows against https://github.com/aspect-build/bazel-examples/tree/main/jest
```

### Any other information?

I had to struggle to get it the example project to build (I needed to disable hooks for two npm dependencies), and also had some issues because the split code/test `ts_project` targets run into the known issue with the sources from one being picked up by the other (I got a lot of file permissions errors when the test compile target tried to produce the same outputs as the source compile target; however even when I changed it to have a single compilation `ts_project` the test target failed to run).

I've tried setting `patch_node_fs = False` (had some problems with test rules on windows before and this seemed to fix it) but didn't seem to help.

The run files tree looks fine - the source and test js files are all output OK as they are on other platforms. I've even added some code to my `jest.config.js` to write some output to a file with the result of calling `glob.sync('**/*.js')` and that picks the files up OK, so they are visible to node...

**Comment by ahmedneilhussain** [comment]
It looks to be an issue with Jest not liking symlinks, or particularly leaf symlinks. Others have raised issues with `bazel` + `jest` before because of this, see https://github.com/jestjs/jest/issues/5356 . This presumably predates `rules_jest` however. On my setup, Bazel 7.1.0 and jest 29.7.0, it worked out of the box on Linux and MacOS but fell over on Windows.

I thought it might have been a `/` `\` path canonicalisation issue, so experimented with increasingly permissive test regex/glob patterns, but it made no difference. However I discovered that if I hacked the run files on my Windows setup and added a real `.js` file rather than a symlink to one, it was picked up.

As discussed in the Jest issue linked above, it is down to the file crawler in `jest-haste-map`. It tries to shell out to `find` if that utility is present, which it is on MacOS and Linux. On Windows, or if `--forceNodeFilesystemAPI` is true (see https://jestjs.io/docs/configuration#haste-object), it crawls the filesystem itself using the Node APIs, and explicitly skips symbolic links (or at least leaf files that are symbolic links). Note that Jest and Node can read the contents of files referenced by symlinks just fine, it's just that without the files being found in the initial crawl, they never get matched: this is why it works so long as `find` is present.

I found that applying the following `pnpm` patch to `jest-haste-map` made it work on Windows:

```
diff --git a/build/crawlers/node.js b/build/crawlers/node.js
index eb90b4e1bce64ab6809d914db9122e1fc8dc151c..e78d3f810f5629d6f9e90d870a162f4b8d59ec60 100644
--- a/build/crawlers/node.js
+++ b/build/crawlers/node.js
@@ -133,9 +133,9 @@ function find(roots, extensions, ignore, enableSymlinks, callback) {
           if (ignore(file)) {
             return;
           }
-          if (entry.isSymbolicLink()) {
-            return;
-          }
+          // if (entry.isSymbolicLink()) {
+          //   return;
+          // }
           if (entry.isDirectory()) {
             search(file);
             return;
```

**Comment by ahmedneilhussain** [comment]
It looks from your build hooks as though you have builds running and passing tests on Windows, but I don't understand how they could unless a port of `find` is present in the environment..? I wondered if there was a Bazel flag (or older Bazel behaviour) that could construct the `runfiles` with full copies rather than links, but couldn't spot anything on skimming the docs.

**Comment by ahmedneilhussain** [comment]
Incidentally https://github.com/aspect-build/rules_jest/blob/9b6a225c210517c8af7a5cfb745d2388f38c02b3/jest/private/jest_config_template.mjs#L95 should perhaps use `??=` rather than `=` so that other haste options are not overwritten/ignored..?

**Comment by jbedard** [comment]
@ahmedneilhussain can you confirm if this is still an issue with latest?
